tests/devices/experimental/test_default_qubit_2.py

```diff
@@ -179,6 +179,17 @@ def test_config_choices_for_adjoint(self):
         assert new_config.use_device_gradient
         assert new_config.grad_on_execution
 
+    @pytest.mark.parametrize("max_workers", [None, 1, 2, 3])
```

Comment:
Can we test with multiproc on the CI? I think we can only have 2 potentially concurrent processes running. Though, I suspect oversubscription isn't an issue here, as long as it returns the correct results.

If we aren't using multiproc on the CI in the below tests, maybe we should try the same parameterisation where applicable.

Reply:
> If we aren't using multiproc on the CI in the below tests, maybe we should try the same parameterisation where applicable.

In places I have an explicit parametrization `@pytest.mark.parametrize("max_workers", [None, 1, 2])` because I need flexibility in device creation, usually to pass in random seeds. Otherwise, I just use the fixture
```
@pytest.fixture(scope="module", params=[None, 1, 2])
def qubit_device_2(request):
    return DefaultQubit2(max_workers=request.param)
```
which is also parametrized over `max_workers`.

> Though, I suspect oversubscription isn't an issue here, as long as it returns the correct results.

The CI is likely oversubscribed, but since the tapes are light it does not seem to cause issues.